### ISSUE TITLE
Ability to assign videos/trailers

### ISSUE THREAD

**Issue by Continous** [JosefNemec/Playnite#405]
Allow the rendering of animated images or videos for use as covers or backgrounds. Preferably with the option to enable hover-over playback only in appearance settings.

**Comment by ArtexJay** [comment]
Yes please! This would be amazing to have. It could be fetching the videos of the game trailers from youtube or something of the sort.

**Comment by DennoCoil** [comment]
Just trying to help, but I was poking around for this and found some helpful things that _might_ be useful.  VLC seems like a good candidate to implement into Playnite as an underlying program.

- It can play practically anything, but there can be some rare problems with it.  One I've commonly found is that videos do tend to have this weird ghosting effect, but that's because there's a missing a key frame in the video and tends to clear up.  It only seems to happen with improperly\weirdly encoded videos or purposefully skipping the video around to various parts.
- It does allow streaming Youtube videos with the support of various formats and outputing to custom Iframes.  https://stackoverflow.com/questions/7582336/force-youtube-embed-to-start-in-720p
- I did find 2 sources on how to bind VLC to WPF and has Python API binds too.  (Do **_not_** use the old Active X plugin because they've stated in their documentation that version is being sunset.)  https://wiki.videolan.org/Documentation:WebPlugin/  https://wiki.videolan.org/Python_bindings

**Comment by Continous** [comment]
The best candidate, in my opinion, is MPV, since it's very bare-bones and designed to be embeddable. It also uses FFMPEG, so support should be essentially universal.

**Comment by DennoCoil** [comment]
Had a quick search and you may be right.  It's pretty much the same thing as VLC, but I had skipped over it because CLI and there's a dozen players out there.

https://www.omgubuntu.co.uk/2017/02/swap-vlc-mpv-ubuntu
https://www.youtube.com/watch?v=Vu-gDXqQ5es

The documentation is also much cleaner.
https://github.com/mpv-player/mpv/wiki
https://mpv.io/manual/stable/#embedding-into-other-programs-libmpv

It would still require a plugin written in C to read from the program, so it would still take a bit of work to implement.

**Comment by JosefNemec** [comment]
There's [Mpv.NET](https://github.com/hudec117/Mpv.NET), but I don't how mature/stable it is.

**Comment by DennoCoil** [comment]
Mpv.NET seems to get regular contributions every other month from the owner, but he does note this on the page:

- Very little documentation has been written for the C API wrapper. Consult client.h.
- The entirety of the mpv C API has not yet been implemented.

It's not the most mature, but MPV is still a stable application.  So it's going to be hit or miss.  At least you won't be reinventing the wheel with a full .NET implementation or writing something in C.

**Comment by ziplock9000** [comment]
It seems my earlier issue from 2017 has been merged with this one. Please remember my suggestion was not just to be able to add videos, but that videos will automatically be pulled from the internet based on automatic searching.

**Comment by gamingexpert13** [comment]
Trailers in the game info section (automatically fetched) is currently the only reason I went back to using GOG Galaxy instead of Playnite. Would be nice to have in playnite. It's the first thing I go to when I'm trying to decide which game to play next.

I'll echo what ziplock9000 said above. I almost submitted a separate feature request because this one didn't really sound like the same thing.

**Comment by erik90mx** [comment]
Same as @gamingexpert13 I have a big library and a lot of times I forget why I have a game in it, the video in the description on GOG Galaxy helps me a lot yo pick one game to play.

**Comment by Mobile313** [comment]
I noticed that VLC was being tossed around quite a bit earlier in this discussion.  I also noticed that some of the existing themes that make use of videos are dependent on those videos being on the hard drive.  Since Playnite is already pulling data from IGDB, it seems like it would be easier to grab a video link from them and simply stream the video (they appear to ultimately reside on youtube for the most part).

In my mind's eye, the video container (iframe/canvas/webview/etc) could be embedded into themes that choose to support it.  The link for videos would be captured whenever you update metadata from IGDB.  Visibility parameters check the DB for an IGDB link, and displays if present or hides if not.  This (in theory) means that the only thing needed would be incorporating the video links into the metadata search, and creating the video container that holds it.
